Use floating/popup windows using vim-floaterm & Add :Lfcd and :Lflcd

sitiom opened the pull request:
Here's what it looks like:
![image](https://user-images.githubusercontent.com/56180050/100401445-70299b00-3094-11eb-945a-7caa04de696d.png)

The BClose dependency is not needed anymore.
Fixes https://github.com/francoiscabrol/ranger.vim/issues/42 as well.
Added :Lfcd and :Lflcd similar to that of [rafaqz/ranger.vim](https://github.com/rafaqz/ranger.vim)

sitiom committed:
Use floating windows using vim-floaterm

sitiom committed:
Update README.md

sitiom committed:
Update README.md

sitiom committed:
Add :Lfcd, :Lflcd & Remove Bclose

ptzz commented:
i haven't used floating windows myself much and mostly prefer non-split views. What is the main advantage of opening lf in a floating window? One disadvantage is that less of lf preview is visible.

sitiom commented:
I guess it is personal preference and it really depends on your workflow. Other than that, It solved some bugs but I actually forgot what those bugs are.

sitiom commented:
> One disadvantage is that less of lf preview is visible.

Although when I open lf on a small split, Less of lf is also seen of course. Popups quite solve those problems.

ptzz commented:
> > One disadvantage is that less of lf preview is visible.
> 
> Although when I open lf on a small split, Less of lf is also seen of course. Popups quite solve those problems.

Good point. In vanilla vim it opens in full vim window also from a split view. Not sure that's easily doable with the neovim terminal? Perhaps a huge floating window for both flavours is the best option.

sitiom commented:
> Perhaps a huge floating window for both flavours is the best option.

Agreed. `FloatermNew` already has a `height` and `width` parameter to set those things.

![image](https://user-images.githubusercontent.com/56180050/107606358-cd36f500-6c70-11eb-971e-a542b18b2c14.png)

Although perhaps it should be up to the user to set it themselves, maybe in a variable like `g:lf_height` and `g:lf_width`.

ptzz commented:
Looks like floaterm has an lf wrapper of its own, have you tried it? https://github.com/voldikss/vim-floaterm/blob/master/autoload/floaterm/wrapper/lf.vim

sitiom commented:
Yep, but it doesn't have more features like this one (Overriding netrw, LfCurrentDirectory
LfWorkingDirectory, etc.). This change overrides that wrapper and adds the functionality there.

sitiom committed:
Fix wrapper `fzf` and `rg` not working on vim8

From https://github.com/voldikss/vim-floaterm/commit/d67f3a734c6badc45e5a836b0d7c1629b1515d86

sitiom committed:
Improve wrapper callback

From https://github.com/voldikss/vim-floaterm/commit/ee60f419d9422ec58ae9bb1182db31b3d8a0949d

sitiom committed:
Use g:lf_width and g:lf_height to set floating window size

sitiom commented:
> Although perhaps it should be up to the user to set it themselves, maybe in a variable like `g:lf_height` and `g:lf_width`.

Oh I forgot to comment, I have now implemented this and everything should be working fine now.

ptzz commented:
Nice! I tested a bit with neovim and MacVim. It seems to work well and I like it. I think this can be merged once the merge conflicts are resolved. Thanks!

sitiom committed:
Merge branch 'master' into floaterm

sitiom commented:
> I think this can be merged once the merge conflicts are resolved. Thanks!

Fixed!

sitiom committed:
Revert "Fix wrapper `fzf` and `rg` not working on vim8"

https://github.com/voldikss/vim-floaterm/issues/249

This reverts commit fd428695cbecfef6ecf7ad9f1497c13b4a10153b.
This change is not needed on lf anyway.
